federation/pkg/federation-controller/util/test/test_helper.go

```diff
@@ -44,77 +44,57 @@ func (wd *WatcherDispatcher) register(watcher *watch.FakeWatcher) {
 	defer wd.Unlock()
 	wd.watchers = append(wd.watchers, watcher)
 	for _, event := range wd.eventsSoFar {
-		go watcher.Action(event.Type, event.Object)
+		watcher.Action(event.Type, event.Object)
```

Comment:
Why are you removing those goroutines everywhere? This seems to better simulate watches...